tests/foreman/api/test_contentviewversion.py

```diff
@@ -33,960 +33,1013 @@
+    _, lce2, _ = module_lce_cv
+    cv = entities.ContentView(organization=module_org).create()
+    # ... and publish it.
+    cv.publish()
+    # Refresh the entity
+    cv = cv.read()
+    # Check that we have a new version
+    assert len(cv.version) == 1
+    version = cv.version[0].read()
+    # The immediate lifecycle is lce1, not lce2
+    with pytest.raises(HTTPError):
+        promote(version, lce2.id)
+
+
+# Tests for content view version promotion.
+
+
+@pytest.mark.tier2
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
+def test_positive_delete(module_org):
+    """Create content view and publish it. After that try to
+    disassociate content view from 'Library' environment through
+    'delete_from_environment' command and delete content view version from
+    that content view. Add repository and gpg key to initial content view
+    for better coverage
+
+    :id: 066dec47-c942-4c01-8956-359c8b23a6d4
+
+    :expectedresults: Content version deleted successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    key_content = read_data_file(ZOO_CUSTOM_GPG_KEY)
+    gpgkey = entities.GPGKey(content=key_content, organization=module_org).create()
+    # Creates new product without selecting GPGkey
+    product = entities.Product(organization=module_org).create()
+    # Creates new repository with GPGKey
+    repo = entities.Repository(gpg_key=gpgkey, product=product, url=FAKE_1_YUM_REPO).create()
+    # sync repository
+    repo.sync()
+    # Create content view
+    content_view = entities.ContentView(organization=module_org).create()
+    # Associate repository to new content view
+    content_view.repository = [repo]
+    content_view = content_view.update(['repository'])
+    # Publish content view
+    content_view.publish()
+    content_view = content_view.read()
+    # Get published content-view version id
+    assert len(content_view.version) == 1
+    cvv = content_view.version[0].read()
+    assert len(cvv.environment) == 1
+    # Delete the content-view version from selected env
+    content_view.delete_from_environment(cvv.environment[0].id)
+    # Delete the version
+    content_view.version[0].delete()
+    # Make sure that content view version is really removed
+    assert len(content_view.read().version) == 0
+
+
+@pytest.mark.upgrade
+@pytest.mark.tier2
+def test_positive_delete_non_default(module_org):
+    """Create content view and publish and promote it to new
+    environment. After that try to disassociate content view from 'Library'
+    and one more non-default environments through 'delete_from_environment'
+    command and delete content view version from that content view.
+
+    :id: 95bb973c-ebec-4a72-a1b6-ad28b66bd11b
+
+    :expectedresults: Content view version deleted successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    content_view = entities.ContentView(organization=module_org).create()
```

Comment:
I see this is being created for most of the tests, I would suggest you to create a function scoped fixture in this module to be used in all the tests.